### DISCUSSION TITLE
Drafts on a separate branch

### DISCUSSION THREAD

**Discussion by markshenouda** [avitorio/outstatic#124] [Ideas]
It's a feature that I liked when I was using Netlify CMS which saving the draft changes on a separate branch not the main branch so your project will be only rebuilt when you publish a new article

**Comment by avitorio** [comment]
Sounds like a good idea. How was this setup in netlify cms? Env vars?

I think we are reaching a point where some of the config vars could probably live in a json file and be edited through the settings tab.

**Comment by markshenouda** [comment]
It's been managed by `publish_mode` in the `config.yml`

https://decapcms.org/docs/configuration-options/#publish-mode

Well the idea is great of having a json file for settings

**Reply by avitorio** [reply]
Thanks for the info. An MVP kind of way to launch this fast would be to make it an env var. `OST_DRAFT_BRANCH` - So that on document save we check if it's a draft and push the change to that specific branch.

My question would be, how do we make sure that we get both published and drafts in the dashboard? 🤔

**Reply by markshenouda** [reply]
Hmmm a good question, it probably fetches both of main and the draft branch (you can name it a default name if the user didn't like `outstatic_draft` or what ever) but pulling the main branch and only the file changes in the draft branch
I don't know how outstatic works behind the scenes but I'll take a look at the code and back to you

**Reply by avitorio** [reply]
Thanks @markshenouda!
Let me know if you need any help!
